### PR TITLE
Switch the lib to A-labels by default

### PR DESCRIPTION
**Pull request by weppos**
Implements the changes described in #31.

The internal representation is now expected to be ASCII. The library will also automatically converts parsed lists from Unicode to ASCII, unless the developers knows the input being ASCII (in this case it's possible to disable the conversion to save processing time).

/cc @jsha @patf @cpu I used the data you provided as test cases, and I should have covered all the needs. Do you mind to review?

It should be safe to run the test suite for letsencrypt/boulder#2278 around this branch. Once confirmed, I'll merge this branch to master and release.

### CODE REVIEW

**Comment by cpu** [comment]
One other minor comment: it would be valuable to update README.md to briefly discuss the IDN support and the default A-label behaviour.

**Comment by weppos** [comment]
Thanks for the review and confirmation @cpu. Also, ❤️  for the kind words.

> One other minor comment: it would be valuable to update README.md to briefly discuss the IDN support and the default A-label behaviour.

That's a fair point. I'll work on it today, and update the branch. Feel free to provide a patch, if you have something specific in mind.

One quick question. As you see from https://github.com/weppos/publicsuffix-go/commit/d5b7003547f2cc7d098662971a370e27aa4a38bc, I've been starting to provide fixed release as I do for other languages, although in Go this is less relevant.

The other main difference is that, while I don't sign each commit, I generally sign each single tag (see https://github.com/weppos/publicsuffix-go/releases/tag/v0.2.0).

Would it be something you may want to rely upon? Eg. pull in only fixed release version, instead of linking directly to master? I'm just checking, for the sake of my curiosity.

**Comment by cpu** [comment]
> That's a fair point. I'll work on it today, and update the branch. Feel free to provide a patch, if you have something specific in mind.

I don't have anything specific in mind, just thought it was a worthy feature to discuss. A small note on the default encoding would probably suffice.

> Would it be something you may want to rely upon? Eg. pull in only fixed release version, instead of linking directly to master? I'm just checking, for the sake of my curiosity.

I would be in favour of switching to using tagged releases instead of master. I've been slowly [cleaning up](https://github.com/letsencrypt/boulder/issues/2315) our dependencies with the goal of moving more towards tagged releases instead of commits. I think it makes reasoning about the dependencies easier overall.

**Comment by weppos** [comment]
> I would be in favour of switching to using tagged releases instead of master. I've been slowly cleaning up our dependencies with the goal of moving more towards tagged releases instead of commits. I think it makes reasoning about the dependencies easier overall.

I'm glad to see we are on the same track. I'll make sure to merge and prep a new release then.

**Comment by weppos** [comment]
I added the requested notes in the README. I think it's ready to merge (feel free to let me know if there are any typo, as I'm not native 🇺🇸  speaker).

Thanks @jsha and @cpu for comments and review.

File: README.md [hunk 1]
```diff
@@ -100,6 +100,19 @@ publicsuffix.DomainFromListWithOptions(list, "google.blogspot.com", nil)
...
 
+## IDN domains, A-labels and U-labels
+
+[A-label and U-label](https://tools.ietf.org/html/rfc5890#section-2.3.2.1) are two different ways to represent IDN domain names. These two encodings are also known as ASCII (A-label) or Pynucode vs Unicode (U-label). Conversions between U-labels and A-labels are performed according to the ["Punycode" specification](https://tools.ietf.org/html/rfc3492), adding or removing the ACE prefix as needed.
+
+IDNA-aware applications generally use the A-label form for storing and manipulating data, whereas the U-labels can appear in presentation and user interface forms.
+
+Although the PSL list has been traditionally U-label encoded, this library follows the common industry standards and stores the rules in their A-label form. Therefore, unless explicitly mentioned, any method call, comparison or internal representation is expected to be ASCII-compatible encoded (ACE).
+
+Passing Unicode names to the library may either result is error or unexpected behaviors.
```
Comment (cpu) [inline README.md]: "may either result is error or" -> "may either result in error or"

Comment (weppos) [inline README.md]: ✔️

**Pull request merged**